### DISCUSSION TITLE
[Feature] Typed album

### DISCUSSION THREAD

**Discussion by axgd-code** [immich-app/immich#6114] [Feature Request]
As an Immich user, I would like to create various types of albums with distinct representations:

- A travel-centric album centered on a map and route, allowing the display of photos based on a plotted path (similar to https://www.polarsteps.com/ or https://findpenguins.com/) with an optional story for each day.
-  A special newborn album oriented around a monthly timeline, showcasing the best photos from each month (akin to https://family-album.com/).
- A wedding album emphasizing different wedding moments  (like https://www.wedshoots.com/).

These albums should accommodate more text and be shareable with the entire family.

**Comment by axgd-code** [comment]
I have other types of albums in mind:
- Christmas centered around the events (the meal, the tree, gifts, etc.)
- Birthday (friends, the party)
- Family album.
An AI could also automatically suggest creating albums based on the type of photos in the library.

**Reply by AgentBignose** [reply]
beeing able to add relations to people would be nice in addition to the birthdays!
With that you are not far from a family tree!

**Comment by plord12** [comment]
Good ideas. 

I would also love to see a map added to an album with gpx support (requested elsewhere)

**Reply by axgd-code** [reply]
I think that this is the discussion https://github.com/immich-app/immich/discussions/6831

**Comment by axgd-code** [comment]
Here is a screenshot showing why Polarsteps' view is awesome: it features a timeline to display the days of the trip/adventure, an image feed, and a map of the trip.
Some public examples:
- https://www.polarsteps.com/fiveontourdstx/5220184-tour-du-monde-en-11-mois
- https://www.polarsteps.com/tripdevoyage/1856667-trip-de-voyage
<img width="1470" alt="Capture d’écran 2024-08-18 à 23 22 14" src="https://github.com/user-attachments/assets/32d14454-8aec-4897-8e98-a0a373433713">



I just discovered this app: https://github.com/seanmorley15/AdventureLog, which seems to achieve the same goal as Polarsteps, I think.
If an app could develop this awesome album viewer, would it be Immich or AdventureLog?

**Reply by floco** [reply]
Great idea ! 
looking for this as well.
As immich has all the photos, the "travel" custom album could show a map of the trip and below day by day what was visited.
Would be better than polarsteps or findmypinguins as no need to use a seperate app to add steps since the photos already have the geolocations and dates/times.

**Comment by Tealk506** [comment]
I was just about to open a new feature request for this exact functionality, but fortunately I found this existing thread.
I would like to add my support for the travel-centric album (similar to Polarsteps).

While the current map feature in Immich works well for general photo discovery, seeing scattered dots on a static map lacks the chronological context of a trip. A dedicated journey view would be a practical addition.

To outline how this specific album type could function:

- Chronological Routing: Use GPS data and timestamps to automatically draw a connected line on the map, displaying the actual path taken over time.
- Smart Clustering & Context: Group photos into specific stops or days along the drawn route. Users should be able to add a text description to these individual clusters/days.
- Interaction: Enable the ability to comment on and like these specific stops/days within the journey album.
- Synced Scrolling: Automatically pan and zoom the map to the corresponding location as the user scrolls through the timeline of photos and text entries.
- Sharing Integration: Utilize the existing sharing mechanisms for these journey albums (internal user sharing, public links, and password-protected links).
- Optional - Subscriptions/Following: Allow users to "follow" a specific journey album and receive notifications when the creator adds a new stop or update.

Currently, users looking for this functionality have to rely on third-party cloud apps like Polarsteps to share travel logs. Having a native option in Immich would be a very useful alternative for self-hosters.

Written with Love, translated by AI
